{{cookiecutter.plugin_name}}/src/{{cookiecutter.module_name}}/_writer.py

```diff
@@ -15,9 +15,15 @@
     FullLayerData = Tuple[DataType, dict, str]
 
 
-def write_single_image(path: str, data: Any, meta: dict):
+def write_single_image(path: str, data: Any, meta: dict) -> List[str]:
     """Writes a single image layer"""
 
```

Comment:
since we're now actually implementing the function body, maybe we should also add a comment above the return value that says something like `# implement your writer logic here ...`

Reply:
complete side note here... @nclack, maybe we should reconsider the "merit" of requiring them to return a list of strings.  Do we do anything with it?  could we instead check `os.stat` of the path to check if anything happened?  or, is there good reason to require them to return a list of paths (e.g. perhaps with remote paths where we can't evaluate before/after)

Reply:
totally agree. Returning the list of strings is awkward.  We don't really do anything with the info and it would be more natural for people to rely on exceptions.

`os.stat` as a check is an idea, but it has some cost and doesn't really do anything that a properly handled write API doesn't do.

Reply:
sounds good, will open npe2 issue to keep discussion going